cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/r4/R4CareGapsProcessor.java

```diff
@@ -0,0 +1,182 @@
+        List<IdType> collectedMeasureIds =
+                measures.stream().map(Resource::getIdElement).collect(Collectors.toList());
+
+        // validate required parameter values
+        checkValidStatusCode(statuses);
+        measureCompatibilityCheck(measures);
+
+        // Subject Population for Report
+        List<String> subjects = getSubjects(subject);
+
+        // Build Results
+        Parameters result = initializeResult();
+
+        // Build Patient Bundles
+
+        List<Parameters.ParametersParameterComponent> components = r4CareGapsBundleBuilder.makePatientBundles(
+                periodStart, periodEnd, subjects, statuses, collectedMeasureIds);
+
+        // Return Results with Bundles
+        return result.setParameter(components);
+    }
+
+    protected List<String> getSubjects(String subject) {
+        R4RepositorySubjectProvider subjectProvider = new R4RepositorySubjectProvider();
+        var subjects = subjectProvider.getSubjects(repository, null, subject).collect(Collectors.toList());
+        if (!subjects.isEmpty()) {
+            ourLog.info(String.format("care-gaps report requested for: %s subjects.", subjects.size()));
+        } else {
+            ourLog.info("care-gaps report requested for: 0 subjects.");
+        }
+        return subjects;
+    }
+
+    private void addConfiguredResource(String id, String key) {
+        // read resource from repository
+        Resource resource = repository.read(Organization.class, new IdType(id));
+
+        // validate resource
+        checkNotNull(
+                resource,
+                String.format(
+                        "The %s Resource is configured as the %s but the Resource could not be read.",
+                        careGapsProperties.getCareGapsReporter(), key));
+
+        // add resource to configured resources
+        configuredResources.put(key, resource);
+    }
+
+    private void checkMeasureImprovementNotation(Measure measure) {
+        if (!measure.hasImprovementNotation()) {
+            ourLog.warn(
+                    "Measure '{}' does not specify an improvement notation, defaulting to: '{}'.",
+                    measure.getId(),
+                    "increase");
+        }
+    }
+
+    private Parameters initializeResult() {
+        return newResource(Parameters.class, "care-gaps-report-" + UUID.randomUUID());
+    }
+
+    private List<String> canonicalToString(List<CanonicalType> measureUrls) {
+        return measureUrls.stream()
+                .map(PrimitiveType::toString)
+                .map(x -> x.replace("CanonicalType[", ""))
+                .map(x -> x.replace("]", ""))
+                .collect(Collectors.toList());
+    }
+
+    private void checkValidStatusCode(List<String> statuses) {
+        r4MeasureServiceUtils.listThrowIllegalArgumentIfEmpty(statuses, "status");
+
+        for (String status : statuses) {
+            if (!status.equals(CareGapsStatusCode.CLOSED_GAP.toString())
```

Comment:
Please invert these:  (ex: ! CareGapsStatusCode.OPEN_GAP.toString().equals(status)).

This is a good habit to have in case the passed in variable ends up being null.